ios/engine/KMEI/KeymanEngine/resources/Keyman.bundle/Contents/Resources/ios-host.js

```diff
@@ -31,7 +31,7 @@ sentryManager.init();
 window.addEventListener('load', init, false);
 
 function init() {
-    const device = navigator.userAgent.indexOf('iPad') >= 0 ? 'AppleTablet' : 'AppleMobile';
+    const device = navigator.userAgent.match(/iPad|Macintosh/) ? 'AppleTablet' : 'AppleMobile';
```

Comment:
Hmm... per #2267 this could cause things to swing the other way and break things on the iPhone.

Reply:
The system setting responsible for this is on by default for tablets and off by default for phones, but the user can set either device type to either behavior.

Reply:
The system setting doesn't affect WKWebView, only Safari. I have verified this on Simulator, iPhone 14 Pro Max, iOS 16.0.

Reply:
The difference in #2267 is that running KeymanWeb in Safari, whereas here we are running Keyman Engine embedded in WKWebView.

Reply:
And the system setting in question, for clarity, is:

![image](https://user-images.githubusercontent.com/4498365/198937441-9e33d289-261c-4ef7-af2c-a56f96e5dbd1.png)

That's correct, yes?

Reply:
Yes.

If memory serves, it makes no difference to the user-agent string whether or not it's in stand-alone Safari or in a WKWebView - that setting will apply to both.  Granted, it has been quite a while since I wrote that PR.

Reply:
... right, I think one of the fixes force-set the embedded WebView such that the toggle would always act as "off" within the app.  I think.

Reply:
Indeed. I am seeing correct behaviour on all 4 segments of the matrix:

x | iPhone | iPad
--|--------|-----
\< 13.0 | ✔ | ✔
\>= 13.0 | ✔ | ✔

We could be specific if we applied the setting [`preferredContentMode`](https://developer.apple.com/documentation/webkit/wkwebpagepreferences/3194426-preferredcontentmode).

Reply:
I haven't fully traced it down yet, but within the description of #2267:

> When run embedded within the iOS app, the app will ensure that a touch layout is used - there's an extra fallback setting that 'saves' us here - but iOS styling will be missing.

Reply:
Well, we don't use `preferredContentMode` anywhere in the source :grin: Perhaps the old `UIWebView` inherited user settings from Safari where `WKWebView` doesn't?

Reply:
... I wouldn't be surprised if `setDefaultDeviceOptions` actually _is_ that something, come to think of it.  In which case... yeah, that check would make sense here.